### ISSUE TITLE
Improvement to Amazon MQ - export instances IP addresses to allow ELB creation

### ISSUE THREAD

**Issue by nestorbolivar** [terraform-providers/terraform-provider-aws#2821]
Hi There,

It would be great to export Amazon MQ instances IDs or IP addresses. We could create ELBs placed in public subnets and redirect them to the admin consoles.

Cheers,
Nestor

**Comment by loivis** [comment]
I think AWS doesn't expose such info, at least not in `DescribeBrokerResponse`? `BrokerInstance` has only attributes as `ConsoleURL` and `Endpoints`, which are already exported.

**Comment by nestorbolivar** [comment]
I get the IP address as follows;

```data "dns_a_record_set" "elb1" {
  count = "${var.enable_aamq == "True" ? 1 : 0}"
  host = "${aws_mq_broker.lp_aamq_brk.id}-1.mq.${var.aws_region}.amazonaws.com"
}```

This os for the target group:

```resource "aws_lb_target_group_attachment" "mq_ip_attach_1" {
  count = "${var.enable_aamq == "True" ? 1 : 0}"
  depends_on = ["aws_mq_broker.lp_aamq_brk"]
  target_group_arn = "${aws_lb_target_group.aamq_elb_tg.arn}"
  target_id        = "${data.dns_a_record_set.elb1.addrs[0]}"
  port             = 8162
}```

**Comment by loivis** [comment]
In you case, I think it's reasonable that terraform ouputs hostname attributes for `mq_broker`. Then you don't need to concatenate strings to get the hostname. But `dns_a_record_set` data source is still need to get ip from hostname. IMPO, I don't think terraform should do this dns lookup.

**Comment by loivis** [comment]
Seems like it's possible to query `eni` to get those IPs. AWS is following the similar rule as `elb` to have mq id within description.

**Comment by bflad** [comment]
The `ip_address` attribute inside `instances` has been exposed via #6103 and released in version 1.40.0 of the AWS provider last week. Please reach out if this is not what you were looking for here.
